Core/include/Acts/AmbiguityResolution/AthenaAmbiguityResolution.ipp

```diff
@@ -0,0 +1,406 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/AmbiguityResolution/AthenaAmbiguityResolution.hpp"
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/EventData/VectorMultiTrajectory.hpp"
+#include "Acts/EventData/VectorTrackContainer.hpp"
+#include "Acts/Utilities/VectorHelpers.hpp"
+
+#include <unordered_map>
+
+namespace Acts {
+
+template <typename track_container_t, typename traj_t,
+          template <typename> class holder_t>
+const TrackContainer<track_container_t, traj_t, holder_t>
+Acts::AthenaAmbiguityResolution::prepareOutputTrack(
+    const TrackContainer<track_container_t, traj_t, holder_t>& tracks,
+    std::vector<std::size_t>& goodTracks) const {
+  auto trackStateContainer = tracks.trackStateContainerHolder();
+  auto trackContainer = std::make_shared<VectorTrackContainer>();
+  trackContainer->reserve(goodTracks.size());
+  // Temporary empty track state container: we don't change the original one,
+  // but we need one for filtering
+  auto tempTrackStateContainer = std::make_shared<VectorMultiTrajectory>();
+  TrackContainer solvedTracks{trackContainer, tempTrackStateContainer};
+  solvedTracks.ensureDynamicColumns(tracks);
+
+  for (auto&& iTrack : goodTracks) {
+    auto destProxy = solvedTracks.getTrack(solvedTracks.addTrack());
+    auto srcProxy = tracks.getTrack(iTrack);
+    destProxy.copyFrom(srcProxy, false);
+    destProxy.tipIndex() = srcProxy.tipIndex();
+  }
+
+  const TrackContainer<track_container_t, traj_t, holder_t> outputTracks{
+      std::make_shared<VectorTrackContainer>(std::move(*trackContainer)),
+      trackStateContainer};
+  return outputTracks;
+}
+
+template <typename track_container_t, typename traj_t,
+          template <typename> class holder_t, typename source_link_hash_t,
+          typename source_link_equality_t>
+std::vector<std::vector<std::tuple<std::size_t, std::size_t, bool>>>
+AthenaAmbiguityResolution::computeInitialState(
+    const TrackContainer<track_container_t, traj_t, holder_t>& tracks,
+    source_link_hash_t&& sourceLinkHash,
+    source_link_equality_t&& sourceLinkEquality) const {
+  auto measurementIndexMap =
+      std::unordered_map<SourceLink, std::size_t, source_link_hash_t,
+                         source_link_equality_t>(0, sourceLinkHash,
+                                                 sourceLinkEquality);
+
+  std::vector<std::vector<std::tuple<std::size_t, std::size_t, bool>>>
+      measurementsPerTrack;
+
+  ACTS_VERBOSE("Starting to compute initial state");
+
+  for (const auto& track : tracks) {
+    std::vector<std::tuple<std::size_t, std::size_t, bool>> measurements_tuples;
```

Comment:
```suggestion
    std::vector<std::tuple<std::size_t, std::size_t, bool>> measurementsTuples;
```

Reply:
snake case has be replaced with camel case